### PR TITLE
added ability to run spectral and display the json output

### PR DESCRIPTION
**Pull request by duckworthj39**
https://cddodatamarketplace.atlassian.net/browse/GDX-173

This pr handles creating a spectral class we can use to output the json from spectral. We also handle installing the spectral dependency, this will need testing on heroku to ensure it works within that environment and we don't have any issues with the system call.

Right now all this pr does is spit out a json response, the next pr will make it look nicer.

### CODE REVIEW

**Review by georgeschena — APPROVED** [review]

File: app/lib/linters/spectral.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Linters
+  class Spectral
+    def initialize(file:)
+      @file = file
+    end
+
+    def lint_to_json
+      `spectral lint -f json "#{file.path}"`
+    end
```
Comment (RobNicholsGDS) [inline app/lib/linters/spectral.rb:9]: If this object is only ever going to be used to return the json derived from the input file, I'd suggest adding a class method:
```ruby
def self.lint_to_json(file:)
  new(file).lint_to_json
end
```
Then the controller line 20 becomes:
```ruby
linter_output = Linters::Spectral.lint_to_json(file: ruleset_params[:oas_file])
```

File: app/lib/linters/spectral.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    private
+
+    attr_reader :file
+  end
+end
```
Comment (RobNicholsGDS) [inline app/lib/linters/spectral.rb:15]: Blank line at end of file need!

Comment (duckworthj39) [reply app/lib/linters/spectral.rb:15]: I need to add the editorconfig in, will do that in this pr!

File: app/lib/linters/spectral.rb (hunk 1 above)
Comment (duckworthj39) [reply app/lib/linters/spectral.rb:9]: Yeah that's a good shout, cheers for that, I'll make that change

**Review by DarrylMultisided — APPROVED** [review]

**Comment by duckworthj39** [comment]
This got merged before I got a chance to make that change so I will do that in the next pr

**Comment by RobNicholsGDS** [comment]
> This got merged before I got a chance to make that change

That's why it is usually not best practice to close and merge someone else's PR.

**Pull request merged**